lib/main.js

```diff
@@ -172,83 +164,93 @@ module.exports = class Router extends EventEmitter {
         };
     }
 
-    postBundleParseCallback() {
-        return (req, res, next) => {
-            body(req, res, {}, (error, bodyObj) => {
-                if (error) {
-                    return next(
-                        boom.boomify(error, {
-                            statusCode: 400,
-                            message: 'Failed parsing postBundle POST-body',
-                        })
-                    );
+    postBundleHandler(type) {
+        assert(
+            ALLOWED_TYPES.includes(type),
+            `Expected type to 1 of ${ALLOWED_TYPES.join(', ')}, but got ${type}`
+        );
+        return async (req, res, next) => {
+            let feedIds;
+            try {
+                const payload = await body(req, res, {});
+                const result = schemas.ids.validate(payload);
+                if (result.error) {
+                    throw result.error;
                 }
-                res.locals.payload = bodyObj;
-                next();
-            });
-        };
-    }
+                feedIds = result.value;
+            } catch (e) {
+                return next(
+                    boom.boomify(e, {
+                        statusCode: 400,
+                        message: 'Invalid POST-body',
+                    })
+                );
+            }
 
-    postBundleValidateCallback() {
-        return (req, res, next) => {
-            schemas.ids.validate(res.locals.payload, (error, payload) => {
-                if (error) {
+            try {
+                const response = await this.bundle({
+                    feedIds,
+                    uri: this.buildUri('bundle', req.headers.host, req.secure),
+                    type,
+                });
+                this.emit(
+                    'request success',
+                    res.locals.track,
+                    req.method,
+                    req.path,
+                    response.file
+                );
+                res.json(response);
+            } catch (e) {
+                this.logger.error('Failed creating bundle', e);
+                if (e.message.includes('No file with name')) {
                     return next(
-                        boom.boomify(error, {
-                            statusCode: 400,
-                            message: 'Invalid POST-body',
+                        boom.boomify(e, {
+                            statusCode: 404,
```

Comment:
Why is this a `404 Not-Found`?

Reply:
This error catches only when the sink file fetch cannot find a file. All other errors fall though. Do you think maybe thats wrong in the context of the whole route? ie. We request a bundle and get a 404 when any one of the feed files to be bundled cannot be found. I think that seems right?

Reply:
What about `409 Conflict`?

Reply:
Or maybe:

`412 Precondition Failed`
> The 412 (Precondition Failed) status code indicates that one or more
>  conditions given in the request header fields evaluated to false when
>  tested on the server.  This response code allows the client to place
>   preconditions on the current resource state (its current
>   representations and metadata) and, thus, prevent the request method
>   from being applied if the target resource is in an unexpected state.

Reply:
Change not related, but happy to change...

Reply:
Does it have to be header fields? Looks like it maybe. In this case we are dealing with an array of hashes sent as a payload and 1 or more of them not being resolvable. Maybe the 409 option would be better?

Reply:
409 it is then? @ivarconr